frontend/src/composables/useCustomColors.js

```diff
@@ -0,0 +1,46 @@
+//
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import { watch } from 'vue'
+import { useTheme } from 'vuetify'
+import vuetifyColors from 'vuetify/lib/util/colors'
+import { toValue } from '@vueuse/core'
+
+import { isHtmlColorCode } from '@/utils'
+
+import { get } from '@/lodash'
+
+export const useCustomColors = (customThemes, theme = useTheme()) => {
+  return new Promise((resolve, reject) => {
+    const unwatch = watch(customThemes, value => {
+      if (value) {
+        clearTimeout(timeoutId)
+        unwatch()
+        for (const colorMode of ['light', 'dark']) {
+          const vuetifyThemeColors = toValue(theme.themes)?.[colorMode]?.colors ?? {}
+          const customThemeColors = value[colorMode] ?? {}
+          for (const [key, value] of Object.entries(customThemeColors)) {
+            if (key in vuetifyThemeColors) {
+              const colorCode = get(vuetifyColors, value)
+              if (colorCode) {
+                vuetifyThemeColors[key] = colorCode
+              } else if (isHtmlColorCode(value)) {
+                vuetifyThemeColors[key] = value
+              }
+            }
+          }
+        }
+        resolve()
+      }
+    }, {
+      immediate: true,
+    })
+    const timeoutId = setTimeout(() => {
+      unwatch()
+      reject(new Error('Setting custom colors timed out'))
```

Comment:
having no theme configured, this error will then be printed to the console

Reply:
Now catching and looging the timeout in the `App` component in case of errors. If no themes are configured the `themes` property is set from `null` to an empty object `{}`. The watch does not time out in this case and no error is thrown.